Sources/StreamChat/StateLayer/ConnectedUser.swift

```diff
@@ -0,0 +1,169 @@
+    public func update(
+        name: String? = nil,
+        imageURL: URL? = nil,
+        extraData: [String: RawJSON] = [:]
+    ) async throws {
+        try await currentUserUpdater.updateUserData(
+            currentUserId: try currentUserId(),
+            name: name,
+            imageURL: imageURL,
+            userExtraData: extraData
+        )
+    }
+    
+    // MARK: - Managing Channels
+    
+    /// Mark all the user's channels as read.
+    ///
+    /// - Throws: An error while communicating with the Stream API or when user is not logged in.
+    public func markAllChannelsRead() async throws {
+        try await currentUserUpdater.markAllRead(currentUserId: try currentUserId())
+    }
+    
+    // MARK: - Managing User Devices
+    
+    /// Loads an array of devices associated with the logged-in user.
+    ///
+    /// - Note: Devices can be read later from ``ConnectedUserState/devices``.
+    ///
+    /// - Throws: An error while communicating with the Stream API or when user is not logged in.
+    /// - Returns: An array of devices receiving push notifications.
+    public func loadDevices() async throws -> [Device] {
+        try await currentUserUpdater.fetchDevices(currentUserId: try currentUserId())
+    }
+    
+    /// Registers the current user's device for push notifications.
+    ///
+    /// Registering a device associates it with the user and tells the push provider to send new message notifications to that device.
+    ///
+    /// - Tip: Register the user's device for remote push notifications once your user is successfully connected to Chat.
+    ///
+    /// - Parameter device: The device information required for registering the device. Use ``PushDevice/apn(token:providerName:)`` for APN.
+    ///
+    /// - Throws: An error while communicating with the Stream API or when user is not logged in.
+    public func addDevice(_ device: PushDevice) async throws {
+        try await currentUserUpdater.addDevice(device, currentUserId: try currentUserId())
+    }
+    
+    /// Removes the specified device from the current user.
+    ///
+    /// Unregistering a device removes the device from the user and stops further new message notifications.
+    ///
+    /// - Parameter deviceId: The id of the device to unregister from push notifications.
+    ///
+    /// - Throws: An error while communicating with the Stream API or when user is not logged in.
+    public func removeDevice(_ deviceId: DeviceId) async throws {
+        try await currentUserUpdater.removeDevice(id: deviceId, currentUserId: try currentUserId())
+    }
+    
+    // MARK: - Moderating Users
+    
+    /// Mutes the user in all the channels.
+    ///
+    /// - Note: Messages from muted users are not delivered via push notifications.
+    ///
+    /// - Parameter userId: The id of the user to mute.
+    ///
+    /// - Throws: An error while communicating with the Stream API.
+    public func muteUser(_ userId: UserId) async throws {
```

Comment:
nit: We have this pattern in other places as well, but the omitting of the label would make sense if we passed the whole `ChatUser`. If we are only passing the ID, then it kind makes sense to still be a named label to make it clear we need to pass an ID 
```suggestion
    public func muteUser(id: UserId) async throws {
```